benchmark/build.gradle.kts

```diff
@@ -0,0 +1,98 @@
+plugins {
+    kotlin("multiplatform")
+    kotlin("plugin.allopen") version "2.0.0"
+    id("org.jetbrains.kotlinx.benchmark") version "0.4.11"
+}
+
+allOpen {
+    // JMH requires all benchmark classes to be open
+    annotation("org.openjdk.jmh.annotations.State")
+}
+
+kotlin {
+    jvmToolchain(11)
+
+    //region JVM Targets
+    jvm()
+    //endregion
+
+    //region JS target
+    js(IR) {
+        nodejs()
+    }
+    //endregion
+
+    //region Native Targets
+    // According to https://kotlinlang.org/docs/native-target-support.html
+    // Tier 1
+    macosX64()
+    macosArm64()
+    iosSimulatorArm64()
+    iosX64()
+
+    // Tier 2
+    linuxX64()
+    linuxArm64()
+    watchosSimulatorArm64()
+    watchosX64()
+    watchosArm32()
+    watchosArm64()
+    tvosSimulatorArm64()
+    tvosX64()
+    tvosArm64()
+    iosArm64()
+
+    // Tier 3
+    mingwX64()
+    //endregion
+
+    sourceSets {
+        commonMain {
+            dependencies {
+                implementation(project.dependencies.platform("com.squareup.okio:okio-bom:3.9.0"))
+                implementation(projects.snakeyamlEngineKmp)
+                implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.4.11")
+                implementation("com.squareup.okio:okio")
+            }
+        }
+
+        jsMain {
+            dependencies {
+                implementation("net.thauvin.erik.urlencoder:urlencoder-lib:1.5.0") {
+                    because("error during benchmark generation: e:\n" +
+                        "KLIB resolver: Could not find \"net.thauvin.erik.urlencoder:urlencoder-lib\" in [.../snakeyaml-engine-kmp]")
+                }
```

Comment:
That's an unexpected error! I wonder why it doesn't happen on other targets. I'll take a look... 

Maybe snake-kmp needs to mark it as `api()`?

Reply:
I think this might be related to how the benchmark plugin resolves dependencies when compiling generated benchmarks. But I believe snake-kmp should not mark this dependency as `api` because it does not expose this library in public API (even if it fixes this problem for some reason)

Reply:
You are actually right... if you declare the dependency to that library as `api`, the benchmark for JS compiles without any issues. But it is not right to include a library as `api` if it is not used in public API. I will raise a question in kotlinx-benchmark repo - maybe they can help with that

Reply:
There is already an issue for that: https://github.com/Kotlin/kotlinx-benchmark/issues/185
`Re-declaration of dependencies.`